### PR TITLE
SearchableMultiSelect: add support for `OptionGroup`

### PR DESCRIPTION
**Pull request by ogupte**
add support for:
 - `OptionGroup`
 - customizable `SelectionLabel`
 - hide options on `isLoading`

## PR Checklist

- Manually tested across supported browsers
  - [x] Chrome
  - [ ] Firefox
  - [ ] Safari
  - [ ] IE11 (Win 7)
  - [ ] Edge (Win 10)
- [x] Unit tests written (`common` at minimum)
- [x] PR has one of the `semver-` labels
- [x] Two core team engineer approvals
- [x] One core team UX approval

http://docspot.devnxs.net/projects/lucid/744-searchablemultiselect-optiongoups-label/#/components/SearchableMultiSelect
screen shot 2017-05-24 at 4 30 39 pm
screen shot 2017-05-24 at 4 31 12 pm

### CODE REVIEW

File: src/components/DropMenu/__snapshots__/DropMenu.spec.jsx.snap [hunk 1]
```diff
@@ -712,7 +710,6 @@ exports[`DropMenu [common] example testing should match snapshot(s) for 09.state
       >
         <div
           className="lucid-DropMenu-label"
-          isHidden={false}
```
Comment (chrisdopuch) [inline src/components/DropMenu/__snapshots__/DropMenu.spec.jsx.snap]: I'm guessing that these changed due to the new omit props?

File: src/components/SearchableMultiSelect/SearchableMultiSelect.jsx [hunk 2]
```diff
@@ -280,48 +321,102 @@ const SearchableMultiSelect = createClass({
...
+							isHidden={!optionFilter(searchText, optionProps)}
+							isDisabled={isLoading}
+						>
+							<div className={cx('&-checkbox')}>
+								<Checkbox
+									onSelect={this.handleCheckboxSelect}
+									callbackId={optionIndex}
+									isSelected={_.includes(selectedIndices, optionIndex)}
+								/>
+								<div className={cx('&-checkbox-label')}>
```
Comment (chrisdopuch) [inline src/components/SearchableMultiSelect/SearchableMultiSelect.jsx]: should this be a `<label>`?

File: src/components/SearchableMultiSelect/SearchableMultiSelect.jsx [hunk 3]
```diff
@@ -407,39 +511,105 @@ const SearchableMultiSelect = createClass({
...
+								}
+							)}
 							{_.map(selectedIndices, selectedIndex => {
-								const optionProps = optionsProps[selectedIndex];
-								const selectionProps = _.get(
-									getFirst(optionProps, SearchableMultiSelect.Option.Selection),
-									'props'
+								const selectedUngroupedOptionData = _.find(
+									ungroupedOptionData,
+									({ optionIndex }) => selectedIndex === optionIndex
```
Comment (chrisdopuch) [inline src/components/SearchableMultiSelect/SearchableMultiSelect.jsx]: /nit `_.find(ungroupedOptionData, {optionIndex: selectedIndex})` shorthand is slightly more concise

File: src/components/DropMenu/__snapshots__/DropMenu.spec.jsx.snap (hunk 1 above)
Comment (ogupte) [inline src/components/DropMenu/__snapshots__/DropMenu.spec.jsx.snap]: correct

**Comment by rsadres** [comment]
This looks fantastic! Behaves as I would expect in all of the examples. 
In the Grouped Options example, can we please double the space under the last object in a group and before the line break for the next group? That will reinforce the section separator.
Thank you!
R

**Pull request merged**